src/payment/strategies/zip/zip-payment-strategy.spec.ts

```diff
@@ -0,0 +1,229 @@
+
+        jest.spyOn(paymentStrategyActionCreator, 'widgetInteraction').mockImplementation(cb => cb());
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+        jest.spyOn(paymentActionCreator, 'submitPayment');
+
+        strategy = new ZipPaymentStrategy(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            paymentMethodActionCreator,
+            paymentStrategyActionCreator,
+            zipScriptLoader
+        );
+
+        jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod');
+    });
+
+    it('creates an instance of ZipPaymentStrategy', () => {
+        expect(strategy).toBeInstanceOf(ZipPaymentStrategy);
+    });
+
+    describe('#initialize()', () => {
+        let zipOptions: PaymentInitializeOptions;
+
+        beforeEach(() => {
+            zipOptions = { methodId: 'zip' };
+        });
+
+        it('loads zip lightbox script', async () => {
+            await strategy.initialize(zipOptions);
+
+            expect(zipScriptLoader.load).toHaveBeenCalled();
+        });
+
+        it('does not load zip if initialization data is not provided', async () => {
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+                .mockReturnValue(undefined);
+            try {
+                await strategy.initialize(zipOptions);
+            } catch (error) {
+                expect(error).toBeInstanceOf(MissingDataError);
+                expect(zipScriptLoader.load).not.toHaveBeenCalled();
+
+            }
+        });
+    });
+
+    describe('#execute()', () => {
+        let zipOptions: PaymentInitializeOptions;
+        let orderRequestBody: OrderRequestBody;
+
+        beforeEach(async () => {
+            orderRequestBody = getOrderRequestBody();
+            zipOptions = { methodId: 'zip' };
+
+            await strategy.initialize(zipOptions);
+        });
+
+        it('dispatch widget interaction when execute is called', async () => {
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(store.dispatch).toHaveBeenCalled();
+        });
+
+        it('expect to get the payment method', async () => {
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(store.getState().paymentMethods.getPaymentMethod).toHaveBeenCalledWith(zipOptions.methodId);
+        });
+
+        it('returns the payment information', async () => {
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(store.getState).toHaveBeenCalled();
+        });
+
+        it('succesfully dispatches widget Interaction', async () => {
+            await strategy.execute(orderRequestBody, zipOptions);
+
+            expect(paymentMethodActionCreator.loadPaymentMethod).toHaveBeenCalledWith('zip');
+            expect(store.getState().paymentMethods.getPaymentMethod).toHaveBeenCalledWith('zip');
```

Comment:
I'm not sure if any of the assertions in this test verifies that widget interaction action is successfully dispatched. Anyway, it doesn't matter as the call is not needed as I explained earlier.